colin/core/checks/labels.py

```diff
@@ -73,3 +77,43 @@ def check(self, target):
                            reference_url=self.reference_url,
                            check_name=self.name,
                            logs=[])
+
+
+class InheritedOptionalLabelAbstractCheck(ImageAbstractCheck):
+
+    def __init__(self, message, description, reference_url, tags):
+        """
+        Abstract check for Dockerfile/Image labels.
+
+        :param message: str
+        :param description: str
+        :param reference_url: str
+        :param tags: [str]
+        """
+        super(InheritedOptionalLabelAbstractCheck, self) \
+            .__init__(message, description, reference_url, tags)
+
+    def check(self, target):
+        optional_labels = [
```

Comment:
I'm also wondering if we want to hardcode this in the codebase; wouldn't it be better to store it in a ruleset?

Reply:
reading it now from ruleset